validator/src/validator.rs

```diff
@@ -257,6 +263,37 @@ impl Validator {
             drop(state);
             self.on_slot_change(SlotChange::NextBlock);
         }
+
+        // If we're a parked validator with a wallet key, we signal to unpark automatically.
+        let validator_registry = NetworkInfo::from_network_id(self.blockchain.network_id).validator_registry_address().expect("Albatross consensus always has the address set.");
+
+        if !self.is_parked(validator_registry) || self.validator_wallet_key.is_none() {
+            return;
+        }
+
+        let mut recipient = Recipient::new_staking_builder(validator_registry.clone());
+        recipient.unpark_validator(&self.validator_key.public);
+
+        // Send the unpark transaction with a fixed height each epoch in case there's already a pushed unpark transaction.
+        // Note: If the validity window is ever less than an epoch's length, using the last macro block's height would require
+        // handling the case where the last macro block is more blocks away than the validity window inverval, since otherwise
+        // invalid transactions would be created.
+        let validity_start_height = policy::macro_block_before(self.consensus.mempool.current_height());
+
+        let mut tx_builder = TransactionBuilder::new();
+        tx_builder.with_sender(validator_registry.clone())
+            .with_value(Coin::ZERO)
+            .with_network_id(self.blockchain.network_id)
+            .with_validity_start_height(validity_start_height)
+            .with_recipient(recipient.generate().unwrap());
+
+        let mut proof_builder = tx_builder.generate().unwrap().unwrap_signalling();
+        proof_builder.sign_with_validator_key_pair(&self.validator_key);
+        let mut proof_builder = proof_builder.generate().unwrap().unwrap_basic();
+        proof_builder.sign_with_key_pair(&self.validator_wallet_key.clone().unwrap());
+        let transaction = proof_builder.generate().unwrap();
+
+        self.consensus.mempool.push_transaction(transaction.clone());
```

Comment:
Minor suggestion here: If we add something to this function in the future, we have to be careful to notice the early return. Thus, I'd replace the return by putting the whole thing into an if. Also, then we can use the more Rusty way – a `if let` statement.
```suggestion
        if let Some(ref validator_wallet_key) = self.validator_wallet_key {
            if self.is_parked(validator_registry) {
                // If we're a parked validator with a wallet key, we signal to unpark automatically.
                let validator_registry = NetworkInfo::from_network_id(self.blockchain.network_id).validator_registry_address().expect("Albatross consensus always has the address set.");

                let mut recipient = Recipient::new_staking_builder(validator_registry.clone());
                recipient.unpark_validator(&self.validator_key.public);

                // Send the unpark transaction with a fixed height each epoch in case there's already a pushed unpark transaction.
                // Note: If the validity window is ever less than an epoch's length, using the last macro block's height would require
                // handling the case where the last macro block is more blocks away than the validity window inverval, since otherwise
                // invalid transactions would be created.
                let validity_start_height = policy::macro_block_before(self.consensus.mempool.current_height());

                let mut tx_builder = TransactionBuilder::new();
                tx_builder.with_sender(validator_registry.clone())
                    .with_value(Coin::ZERO)
                    .with_network_id(self.blockchain.network_id)
                    .with_validity_start_height(validity_start_height)
                    .with_recipient(recipient.generate().unwrap());

                let mut proof_builder = tx_builder.generate().unwrap().unwrap_signalling();
                proof_builder.sign_with_validator_key_pair(&self.validator_key);
                let mut proof_builder = proof_builder.generate().unwrap().unwrap_basic();
                proof_builder.sign_with_key_pair(validator_wallet_key);
                let transaction = proof_builder.generate().unwrap();

                self.consensus.mempool.push_transaction(transaction.clone());
            }
        }
```